welcome message editing!

heronhaye opened the pull request.

jakob223 commented on shared/teams/edit-team-welcome-message/index.tsx:
```diff
@@ -0,0 +1,125 @@
+    throw new Error(`There was a problem loading the welcome message page, please report this error.`)
+  }
+
+  const waitingKey = Container.useSelector(_ => Constants.setWelcomeMessageWaitingKey(teamID))
```
```suggestion
  const waitingKey = Constants.getWelcomeMessageWaitingKey(teamID)
```

jakob223 replied:
`useSelector` is only for when you need stuff from the store.

jakob223 commented on shared/teams/edit-team-welcome-message/index.tsx:
```diff
@@ -0,0 +1,125 @@
+    if (message) {
+      return message
+    }
+    throw new Error(`There was a problem loading the welcome message, please report this error.`)
```
What would cause us to not load this? IMO this is extra code for a case that seems pretty unlikely and would be noticeable anyway (by the welcome message not populating).

heronhaye replied:
I don't think that's possible, because you have to be on the settings tab and the Edit button won't load until it's populated. Originally I had `message!` but wasn't sure if that's safe to do as far as defensive coding goes.

jakob223 replied:
If you don't think it's possible, I think that `message!` seems perfectly acceptable to me.

jakob223 commented on shared/teams/edit-team-welcome-message/index.tsx:
```diff
@@ -0,0 +1,125 @@
+          {welcomeMessage.set && welcomeMessage.text.length === 0 ? (
+            'No welcome note will be shown to new members.'
+          ) : (
+            <>&nbsp;</>
```
why this rather than not showing the `Kb.Text` if the welcome message is set?

heronhaye replied:
So the space under the text box doesn't change when the message doesn't appear.

jakob223 replied:
okay, in that case I think I'd do `? (blah) : (' ')` - if you wrap it in a fragment (`<>` / `</>`) like that you might have issues on mobile

jakob223 commented on shared/teams/team/settings-tab/index.stories.tsx:
```diff
@@ -10,6 +10,7 @@ const commonProps = {
   ignoreAccessRequests: true,
   isBigTeam: true,
   loadWelcomeMessage: Sb.action('loadWelcomeMessage'),
+  onEditWelcomeMessage: () => {},
```
```suggestion
  onEditWelcomeMessage: Sb.action('editWelcomeMessage'),
```

This way when you click the button the storyboard gives you a nice indicator of what was called

heronhaye commented:
@jakob223 I added two commits. First one is a refactor because I forgot that service needs to do some link parsing on the rendered output. Second one (hopefully) addresses all of your feedback.

heronhaye commented:
tagging @mmaxim for chat code changes

heronhaye commented:
Rebased this to master

jakob223 commented:
oops sorry will test this on mobile today and then hopefully approve
